### PR TITLE
dfflibmap: Add a -dont_use flag to ignore cells

### PR DESCRIPTION
**Pull request by rovinski**
This is an alternative to setting the `dont_use` property in lib. This brings `dfflibmap` in parity with the abc pass for `-dont_use`.

This feature is extremely useful in the case where Liberty files are not easily modifiable, such as when they are stored on a read-only network drive.

This is my first PR here, so I am very open to feedback!

### CODE REVIEW

**Comment by rovinski** [comment]
I (hopefully) fixed compilation on Windows and also added a simple test case.

**Comment by QuantamHD** [comment]
@nakengelhardt This one would be useful to us, and @rovinski  is a known collaborator with the OpenROAD project..

**Comment by rovinski** [comment]
Windows build issue resolved - all tests pass now.

File: passes/techmap/dfflibmap.cc [hunk 1]
```diff
@@ -115,7 +122,15 @@ static bool parse_pin(LibertyAst *cell, LibertyAst *attr, std::string &pin_name,
 	return false;
 }
 
-static void find_cell(LibertyAst *ast, IdString cell_type, bool clkpol, bool has_reset, bool rstpol, bool rstval)
+static int glob_match(const char *pattern, const char *string) {
```
Comment (povik) [inline passes/techmap/dfflibmap.cc:125]: Please use `patmatch` from `kernel/yosys.cc` for this, if possible

Comment (rovinski) [reply passes/techmap/dfflibmap.cc:125]: I didn't see that, thanks. Opened #4223.

**Pull request merged**